node/modules/lp2p/rcmgr.go

```diff
@@ -111,10 +113,12 @@ func ResourceManager(connMgrHi uint) func(lc fx.Lifecycle, repo repo.LockedRepo)
 		if err != nil {
 			return nil, fmt.Errorf("error creating resource manager stats reporter: %w", err)
 		}
-		err = view.Register(obs.DefaultViews...)
-		if err != nil {
-			return nil, fmt.Errorf("error registering rcmgr metrics: %w", err)
-		}
+
+		// TODO: Libp2p converted to native prometheus metrics -- how do we register them
```

Comment:
Looks like you're using the [default prometheus register](https://github.com/filecoin-project/lotus/blob/4588523de72452355fb6338176dbab6e69c5853a/cmd/lotus-stats/main.go#L159). You should be able to just do: https://github.com/libp2p/go-libp2p/blob/master/dashboards/resource-manager/README.md

e.g. `rcmgrObs.MustRegisterWith(prometheus.DefaultRegisterer)`

Reply:
Done.

Reply:
Actually, that doesn't work. It makes the tests panic, presumable when multiple hosts are started: https://app.circleci.com/pipelines/github/filecoin-project/lotus/27088/workflows/e4e8800d-6d83-4e2f-950f-3c619de1866a/jobs/874717/steps

Reply:
ah right during a test you may spin up multiple of these.

Reply:
I added a `sync.Once`. Maybe not the nicest solution, but this should help us unblock this PR.